cpp/oneapi/dal/algo/decision_forest/common.hpp

```diff
@@ -0,0 +1,219 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/data/table.hpp"
+#include "oneapi/dal/detail/common.hpp"
+
+namespace oneapi::dal::decision_forest {
+
+namespace detail {
+struct tag {};
+class descriptor_impl;
+class model_impl;
+} // namespace detail
+
+namespace task {
+struct classification {};
+struct regression {};
+using by_default = classification;
+} // namespace task
+
+namespace method {
+struct default_dense {};
+struct hist {};
+using by_default = default_dense;
+} // namespace method
+
+enum class variable_importance_mode
+{
+    none,      /* Do not compute */
+    mdi,       /* Mean Decrease Impurity.
+                       Computed as the sum of weighted impurity decreases for all nodes where the variable is used,
+                       averaged over all trees in the forest */
+    mda_raw,   /* Mean Decrease Accuracy (permutation importance).
+                       For each tree, the prediction error on the out-of-bag portion of the data is computed
+                       (error rate for classification, MSE for regression).
+                       The same is done after permuting each predictor variable.
+                       The difference between the two are then averaged over all trees. */
+    mda_scaled /* Mean Decrease Accuracy (permutation importance).
+                       This is MDA_Raw value scaled by its standard deviation. */
+};
+
+enum class result_to_compute_id
+{
+    compute_out_of_bag_error                 = 0x00000001ULL,
```

Comment:
I think, we can avoid assigning particular values to enum items.

Reply:
in this case it is required to use it in bit operations with results_to_compute value